core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java

```diff
@@ -1404,6 +1404,24 @@ public static Double cosineSimilarity(
         return dotProduct / (normLeftMap * normRightMap);
     }
 
+    @Description("Calculates the cosine distance between the give sparse vectors")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double cosineDistance(
+                    @OperatorDependency(
+                    operator = IDENTICAL,
+                    argumentTypes = {"varchar", "varchar"},
+                    convention = @Convention(arguments = {BLOCK_POSITION, BLOCK_POSITION}, result = NULLABLE_RETURN)) BlockPositionIsIdentical varcharIdentical,
+            @OperatorDependency(
+                    operator = HASH_CODE,
+                    argumentTypes = "varchar",
+                    convention = @Convention(arguments = BLOCK_POSITION, result = FAIL_ON_NULL)) BlockPositionHashCode varcharHashCode,
+            @SqlType("map(varchar,double)") SqlMap leftMap,
+            @SqlType("map(varchar,double)") SqlMap rightMap)
+    {
+        return 1.0 - cosineSimilarity(varcharIdentical, varcharHashCode, leftMap, rightMap);
```

Comment:
The problem is cosineSimilarity can return null when mapL2Norm returns null, which happens when a value in the map it null.  This code should assign cosineSimilarity to a null, and in that case return null... so both this method needs to be nullable also.

Reply:
Updated now .. I hope I got it how you wanted .. test is passing.